examples/pytorch/rgcn/experimental/entity_classify_dist.py

```diff
@@ -0,0 +1,536 @@
+            low_mem=self.low_mem, dropout=self.dropout))
+        # h2h
+        for idx in range(self.num_hidden_layers):
+            self.layers.append(RelGraphConv(
+                self.h_dim, self.h_dim, self.num_rels, "basis",
+                self.num_bases, activation=F.relu, self_loop=self.use_self_loop,
+                low_mem=self.low_mem, dropout=self.dropout))
+        # h2o
+        self.layers.append(RelGraphConv(
+            self.h_dim, self.out_dim, self.num_rels, "basis",
+            self.num_bases, activation=None,
+            self_loop=self.use_self_loop,
+            low_mem=self.low_mem))
+
+    def forward(self, blocks, feats, norm=None):
+        if blocks is None:
+            # full graph training
+            blocks = [self.g] * len(self.layers)
+        h = feats
+        for layer, block in zip(self.layers, blocks):
+            block = block.to(self.device)
+            h = layer(block, h, block.edata['etype'], block.edata['norm'])
+        return h
+
+def init_emb(shape, dtype):
+    arr = th.zeros(shape, dtype=dtype)
+    nn.init.uniform_(arr, -1.0, 1.0)
+    return arr
+
+class DistEmbedLayer(nn.Module):
+    r"""Embedding layer for featureless heterograph.
+    Parameters
+    ----------
+    dev_id : int
+        Device to run the layer.
+    g : DistGraph
+        training graph
+    num_of_ntype : int
+        Number of node types
+    embed_size : int
+        Output embed size
+    sparse_emb: bool
+        Whether to use sparse embedding
+        Default: False
+    dgl_sparse_emb: bool
+        Whether to use DGL sparse embedding
+        Default: False
+    embed_name : str, optional
+        Embed name
+    """
+    def __init__(self,
+                 dev_id,
+                 g,
+                 num_of_ntype,
+                 embed_size,
+                 sparse_emb=False,
+                 dgl_sparse_emb=False,
+                 embed_name='node_emb'):
+        super(DistEmbedLayer, self).__init__()
+        self.dev_id = dev_id
+        self.num_of_ntype = num_of_ntype
+        self.embed_size = embed_size
+        self.embed_name = embed_name
+        self.sparse_emb = sparse_emb
+
+        if sparse_emb:
+            if dgl_sparse_emb:
+                self.node_embeds = dgl.distributed.DistEmbedding(g.number_of_nodes(),
+                                                                 self.embed_size,
+                                                                 embed_name,
+                                                                 init_emb)
+            else:
+                self.node_embeds = th.nn.Embedding(g.number_of_nodes(), self.embed_size, sparse=self.sparse_emb)
+                nn.init.uniform_(self.node_embeds.weight, -1.0, 1.0)
+        else:
+            self.node_embeds = th.nn.Embedding(g.number_of_nodes(), self.embed_size)
+            nn.init.uniform_(self.node_embeds.weight, -1.0, 1.0)
```

Comment:
do we need to provide an option of using dense embedding?

Reply:
Dense can also work in this dataset.
This is an example, we can show different choices.